scaleway/resource_iam_application_test.go

```diff
@@ -0,0 +1,120 @@
+package scaleway
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	iam "github.com/scaleway/scaleway-sdk-go/api/iam/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func init() {
+	resource.AddTestSweepers("scaleway_iam_application", &resource.Sweeper{
+		Name: "scaleway_iam_application",
+		F:    testSweepIamApplication,
+	})
+}
+
+func testSweepIamApplication(_ string) error {
+	return sweep(func(scwClient *scw.Client) error {
+		api := iam.NewAPI(scwClient)
+
+		listApps, err := api.ListApplications(&iam.ListApplicationsRequest{})
+		if err != nil {
+			return fmt.Errorf("failed to list applications: %w", err)
+		}
+		for _, app := range listApps.Applications {
+			err = api.DeleteApplication(&iam.DeleteApplicationRequest{
+				ApplicationID: app.ID,
+			})
+			if err != nil {
+				return fmt.Errorf("failed to delete application: %w", err)
+			}
+		}
+		return nil
+	})
+}
+
+func TestAccScalewayIamApplication_Basic(t *testing.T) {
```

Comment:
Nit: add TestAccScalewayIamApplication_NoUpdate with auto-generated name and 2 steps with same config